server/server.go

```diff
@@ -1415,13 +1402,19 @@ func (s *BgpServer) DeleteBmp(c *config.BmpServerConfig) error {
 func (s *BgpServer) Shutdown() {
 	s.mgmtOperation(func() error {
 		s.shutdown = true
-		stateOp := AdminStateOperation{ADMIN_STATE_DOWN, nil}
+		stateOp := AdminStateOperation{
+			State:         ADMIN_STATE_DOWN,
+			Communication: nil,
+		}
 		for _, p := range s.neighborMap {
 			p.fsm.adminStateCh <- stateOp
 		}
-		// the main goroutine waits for peers' goroutines to stop but if no peer is configured, needs to die immediately.
-		if len(s.neighborMap) == 0 {
-			os.Exit(0)
+		// Waits for all goroutines per peer to stop
+		for _, p := range s.neighborMap {
```

Comment:
A suggestion maybe as a followup would be to have the `Peer` have a `Close` method that can be used to signal the shutdown or server `Stop`.